tests/integration/run.bats

```diff
@@ -159,6 +159,12 @@ function teardown() {
 	[ "$status" -eq 0 ]
 	[[ "$output" = *"Hello World"* ]]
 	[[ "$output" = *"runc-dmz: using /proc/self/exe clone"* ]]
+	if [ "$EUID" -eq 0 ] && is_kernel_gte 5.1 && grep -qFw overlay /proc/filesystems; then
+		# If the kernel has fsopen() and we have privileges to use it, we will
+		# use a temporary overlayfs instead of making a memfd clone of
+		# /proc/self/exe.
+		[[ "$output" = *"runc-dmz: using overlayfs for sealed /proc/self/exe"* ]]
+	fi
```

Comment:
This doesn’t seem to work when running as the UID 0 in a userNS with a vanilla kernel 5.1-5.10 that lacks support for mounting overlayfs in a userNS

Reply:
Right, but does anyone run the whole test suite inside a userns? I can change this test to check for that, but runc it self will gracefully fall back to doing memfds in that case.